### ISSUE TITLE
New Check: Last TestStep Cannot Include ExpectError

### ISSUE THREAD

**Issue by bflad** [bflad/tfproviderlint#162]
## Description

The Terraform Plugin SDK acceptance testing framework can only destroy resources if the final configuration in `helper/resource.TestStep` can be applied successfully. This check can have false positives if the expected error is due an API operation (e.g. resource update), however this might be good to require provider developers to explicitly ignore the check in that case.

## Flagged Code

```go
&resource.TestCase{
  // ...
  Steps: []*resource.TestStep{
    {
      Config: "...",
      Check:  /* ... */,
    },
    {
      Config:      "...",
      ExpectError: /* ... */,
    },
  },
}
```

## Passing Code

```go
&resource.TestCase{
  // ...
  Steps: []*resource.TestStep{
    {
      Config:      "...",
      ExpectError: /* ... */,
    },
    {
      Config: "...",
      Check:  /* ... */,
    },
  },
}
```
